sys/proc.c

```diff
@@ -3,23 +3,26 @@
 #include <proc.h>
 #include <pool.h>
 #include <thread.h>
-#include <sysinit.h>
 #include <klog.h>
 #include <errno.h>
 #include <filedesc.h>
 #include <wait.h>
 #include <signal.h>
 #include <sleepq.h>
 #include <sched.h>
+#include <malloc.h>
 
 static POOL_DEFINE(P_PROC, "proc", sizeof(proc_t));
+static POOL_DEFINE(P_PGRP, "pgrp", sizeof(pgrp_t));
 
 static mtx_t *all_proc_mtx = &MTX_INITIALIZER(0);
 
 /* proc_list, zombie_list and last_pid must be protected by all_proc_mtx */
 static proc_list_t proc_list = TAILQ_HEAD_INITIALIZER(proc_list);
 static proc_list_t zombie_list = TAILQ_HEAD_INITIALIZER(zombie_list);
 
+static pgrp_list_t pgrp_list = LIST_HEAD_INITIALIZER(pgrp_list);
```

Comment:
LIST vs TAILQ?
TAILQ has been already used in this file.

Reply:
They say `LIST_HEAD (3)` that lists are faster and simpler. They seem to be sufficient for process groups.
@cahirwpz wdyt?

Reply:
Looking at `queue.h` I wouldn't say the differences are significant.
`TAILQ` supports appending in _O(1)_, while `LIST` do not.

Reply:
done, changed to TAILQ